src/worker/worker/collectors/misp_collector.py

```diff
@@ -234,27 +208,39 @@ def get_story(self, event: dict, source: dict) -> dict:
             raise RuntimeError("The Taranis event is malformed and does not contain the required properties")
         return self.to_story_dict(story_properties, story_news_items, event.get("Event", {}).get("uuid"))
 
+    def get_taranis_event_ids(self, misp) -> set:
+        # Note: searching here directly for objects that belong to a sharing group by id using the sharinggroup parameter does not work in 2.5.2.
+        # It seems to completely ingnore the sharinggroup parameter and return all objects. The objects controller is poorly documented.
+        taranis_objects: list[MISPObject] = misp.search(controller="objects", object_name="taranis-news-item", pythonify=True)  # type: ignore
 
-if __name__ == "__main__":
-    collector = MISPCollector()
-    source = {
-        "description": "",
-        "icon": None,
-        "id": "b583f4ae-7ec3-492a-a36d-ed9cfc0b4a28",
-        "last_attempted": None,
-        "last_collected": None,
-        "last_error_message": None,
-        "name": "https",
-        "parameters": {
-            "ADDITIONAL_HEADERS": "",
-            "API_KEY": "bXSZEtpNQL6somSCz08x3IzEnDx1bkM6wwZRd0uZ",
-            "PROXY_SERVER": "",
-            "REFRESH_INTERVAL": "",
-            "URL": "https://localhost",
-            "USER_AGENT": "",
-            # "SHARING_GROUP_ID": "1",
-        },
-        "state": -1,
-        "type": "misp_connector",
-    }
-    collector.collect(source)
+        logger.debug(f"{taranis_objects=}")
+        event_ids = set()
+        for obj in taranis_objects:
+            event_ids.add(obj.event_id)  # type: ignore
+            print(f"Object ID: {obj.id}, Event ID: {obj.event_id}, Object Name: {obj.name}")  # type: ignore
+        return event_ids
+
+    def get_taranis_story_dicts(self, events, source) -> list[dict]:
+        story_dicts = [
+            self.get_story(event, source)
+            for event in events
+            if (not self.sharing_group_id or str(event.get("Event", {}).get("sharing_group_id")) == str(self.sharing_group_id))
+        ]
+        return [s for s in story_dicts if s is not None]
```

Comment:
Why not include the None check in the first loop to avoid looping twice?

```python
    return [
        story
        for event in events
        if (
            not self.sharing_group_id
            or str(event.get("Event", {}).get("sharing_group_id")) == str(self.sharing_group_id)
        )
        and (story := self.get_story(event, source)) is not None
    ]
```